sql/driver/conn.go

```diff
@@ -139,3 +140,14 @@ func (c *conn) send(args Request) (*rows, error) {
 	}
 	return r, nil
 }
+
+// Execute all the URL settings against the db to create
+// the correct session state.
+func (c *conn) applySettings(q url.Values) error {
+	if db := q["database"]; len(db) > 0 {
+		if _, err := c.Exec("SET DATABASE = "+db[0], []driver.Value{}); err != nil {
```

Comment:
`db[0]` is because `url.Values` is a `map[string][]string` (`x-www-form-urlencoded` is a silly format). We may want to convert from `url.Values` to a `map[string]string` after parsing the url, and return an error if any keys have more than one value.